targets/unity-v2/Testing/Tests/Client/ClientMultiUserTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿#if !DISABLE_PLAYFABCLIENT_API
+using System;
+using PlayFab.ClientModels;
+
+namespace PlayFab.UUnit
+{
+    public class ClientMultiUserTests : UUnitTestCase
+    {
+        private Action _tickAction;
+
+        private PlayFabAuthenticationContext _player1;
+        private PlayFabAuthenticationContext _player2;
+
+        public override void Tick(UUnitTestContext testContext)
+        {
+            if (_tickAction != null)
+                _tickAction();
+        }
+
+        private void SharedErrorCallback(PlayFabError error)
+        {
+            // This error was not expected.  Report it and fail.
+            ((UUnitTestContext) error.CustomData).Fail(error.GenerateErrorReport());
+        }
+
+        /// <summary>
+        /// CLIENT API
+        /// Test for more then one user can login at the same time asynchronously 
+        /// </summary>
+        [UUnitTest]
+        public void AsyncMultiUserLogin(UUnitTestContext testContext)
+        {
+            var player1LoginRequest = new LoginWithCustomIDRequest
+            {
+                CustomId = PlayFabSettings.BuildIdentifier + "0",
+                CreateAccount = true
+            };
+
+            var player2LoginRequest = new LoginWithCustomIDRequest
+            {
+                CustomId = PlayFabSettings.BuildIdentifier + "1",
+                CreateAccount = true
+            };
+
+            PlayFabClientAPI.LoginWithCustomID(player1LoginRequest, x => _player1 = x.AuthenticationContext,
+                PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+            PlayFabClientAPI.LoginWithCustomID(player2LoginRequest, x => _player2 = x.AuthenticationContext,
+                PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+
+            _tickAction = () =>
+            {
+                // return if players not loaded yet
+                if (_player1 == null || _player2 == null) return;
+
+                // reset delegate to avoid calling this method again
+                _tickAction = delegate { };
+
+                // players must not be equals
+                testContext.False(_player1 == _player2);
```

Comment:
You're testing that the object references are different here, which doesn't actually confirm that they're different players.
You should be testing that _player1.PlayFabId is different from _player2.PlayFabId